commercetools-sdk-base/src/test/java/io/sphere/sdk/queries/PagedQueryResultTest.java

```diff
@@ -38,12 +40,47 @@ public void firstOfManyResult() throws Exception {
 
     @Test
     public void middleOfManyResult() throws Exception {
-        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of((long) PAGE_SIZE, TOTAL, listOfSize(PAGE_SIZE));
+        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of( (long)PAGE_SIZE, TOTAL, listOfSize(PAGE_SIZE));
         assertThat(queryResult.isFirst()).isFalse();
         assertThat(queryResult.isLast()).isFalse();
     }
 
     @Test
+    public void pagesIndexingResults() throws Exception {
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(0L, 0l, listOfSize(0));
+            assertThat(queryResult.getPageIndex()).isEqualTo(0);
+            assertThat(queryResult.getTotalPages()).isEqualTo(0);
+
+        }
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(0L, 106l, listOfSize(25));
+            assertThat(queryResult.getPageIndex()).isEqualTo(0);
+            assertThat(queryResult.getTotalPages()).isEqualTo(5);
+        }
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(75L, 106l, listOfSize(25));
+            assertThat(queryResult.getPageIndex()).isEqualTo(3);
+            assertThat(queryResult.getTotalPages()).isEqualTo(5);
+        }
+
+        {
+            final PagedQueryResult<Integer> queryResult = PagedQueryResult.of(100L, 106l, listOfSize(25));
```

Comment:
This is not a valid answer, because if you are putting an offset of 100 and you have a total of 106 products, then the size of the results you are going to receive is 6. Then with the current implementation you would receive 0, which is wrong. Applying the solution I proposed in the other comment would not return anything (absent) and it would be one of those situations where we can't tell the page structure.